stable/redis-ha/templates/redis-master.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  labels:
+    name: {{ template "fullname" . }}-master
+    redis-sentinel: "true"
+    redis-node: "true"
+    role: master
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+  name: redis-master
+spec:
+  serviceName: "redis"
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: redis
+        redis-sentinel: "true"
+        redis-node: "true"
+        role: master
+        chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+        release: "{{ .Release.Name }}"
+        heritage: "{{ .Release.Service }}"
+    spec:
+      terminationGracePeriodSeconds: 10
+      containers:
+        - name: master
+          image: {{ .Values.redis_image }}
+          env:
+            - name: MASTER
+              value: "true"
+          ports:
+            - containerPort: 6379
+          resources:
+            limits:
+              cpu: "0.1"
+          volumeMounts:
+            - mountPath: /redis-master-data
+              name: data
+        - name: sentinel
```

Comment:
Is there a benefit gained by making a sentinel a sidecar of the initial redis master?

The [comment on this SO response](https://stackoverflow.com/questions/31143072/redis-sentinel-vs-clustering/33027853#33027853) states that there's a redis bug where Sentinel will fail to initiate an election that is resolved by moving sentinels off the redis server.